src/openzaak/components/catalogi/models/mixins.py

```diff
@@ -191,6 +198,12 @@ class Meta:
 
     def publish(self):
         self.concept = False
+        if getattr(self, "_has_overlap", False):
+            self.concept = True  # So it displays concept instead of the date
+            raise ValidationError(
+                f"{self._meta.verbose_name} versies (dezelfde omschrijving) mogen geen "
+                "overlappende geldigheid hebben."
+            )
```

Comment:
I'm not sure I like leaky abstraction of GeldigheidMixin here 

```suggestion
      self.clean()
```